lib/renderHerman.js

```diff
@@ -0,0 +1,155 @@
+'use strict';
+
+const extend = require('extend');
+const path = require('path');
+const Promise = require('bluebird');
+const tinycolor = require('tinycolor2');
+
+const copy = require('./utils/assets');
+const parse = require('./utils/parse');
+const render = require('./utils/render');
+const { nunjucksEnv, templates } = require('./utils/templates');
+
+module.exports = (dest, ctx) => {
+  const indexDest = path.join(dest, 'index.html');
+  const assetsSrc = path.resolve(__dirname, '../dist');
+
+  // Accepts a color (in any format) and returns an object with hex, rgba, and
+  // hsla strings.
+  nunjucksEnv.addFilter('colors', input => {
+    const color = tinycolor(input);
+    if (!color.isValid()) {
+      return null;
+    }
+    const obj = {};
+    const formats = ctx.herman.displayColors || ['hex', 'rgb', 'hsl'];
+    formats.forEach(format => {
+      let fn;
+      switch (format) {
+        case 'hex':
+          fn = 'toHexString';
+          break;
+        case 'rgb':
+          fn = 'toRgbString';
+          break;
+        case 'rgba':
+          format = 'rgb';
+          fn = 'toRgbString';
+          break;
+        case 'hsl':
+          fn = 'toHslString';
+          break;
+        case 'hsla':
+          format = 'hsl';
+          fn = 'toHslString';
+          break;
+      }
+      if (fn) {
+        obj[format] = color[fn]();
+      }
+    });
+    return obj;
+  });
+
+  dest = path.resolve(dest);
+
+  // set (external) base path for links
+  ctx.basePath = '';
+  ctx.activeProject = null;
+  ctx.topGroups = ctx.groups;
+  ctx.topByGroup = ctx.byGroup;
+
+  // check if we need to copy a favicon file or use the default
+  let copyShortcutIcon = false;
+  if (!ctx.shortcutIcon) {
+    ctx.shortcutIcon = { type: 'internal', url: 'assets/img/favicon.ico' };
+  } else if (ctx.shortcutIcon.type === 'internal') {
+    ctx.shortcutIcon.url = `assets/img/${ctx.shortcutIcon.url}`;
+    copyShortcutIcon = true;
+  }
+
+  // render the index template and copy the static assets.
+  const promises = [
+    render(nunjucksEnv, templates.index, indexDest, ctx),
+    copy(
+      path.join(assetsSrc, '/**/*.{css,js,ico,map}'),
+      path.join(dest, 'assets')
+    ).then(() => {
+      if (copyShortcutIcon) {
+        return copy(ctx.shortcutIcon.path, path.resolve(dest, 'assets/img/'));
+      }
+      return Promise.resolve();
```

Comment:
We're already in a `then`; do we need to return a Promise?

Reply:
Yeah, I think it wouldn't strictly be required. But since we're conditionally returning a promise, for consistency I like returning a promise in either case.